src/Config.hs

```diff
@@ -20,16 +20,18 @@ data BlockFetcherConfig = BlockFetcherConfig
   , cfgFetcherFilterJson :: Maybe LBS.ByteString
   , cfgFetcherUseLatest :: Bool
   }
-  deriving stock (Show)
+  deriving stock (Show, Eq)
 
 data Config = Config
   { cfgDbConnectionString :: ByteString
   , cfgServerPort :: Int
+  , -- |if Nothing -- grant the full access
+    cfgServerControlApiToken :: Maybe String
```

Comment:
Security concern, default behaviour should be to deny, and you explicitly must allow all. Otherwise, forgetting to add key may be disastrous.